src/views/components/search.html

```diff
@@ -66,6 +79,16 @@
 				page: 0,
 				query: '',
 				packages: [],
+				type: '',
+				onFilterItemClick( filter ) {
```

Comment:
This isn't really a data function, so it should be on the component itself (same level as `scrollToSearch()`) and called as `on-click="@this.onFilterItemClick('esm')"`

Reply:
So you mean `onFilterItemClick `function should be defined inside `onrender ` function?`

Reply:
no, in the parent object, where `onrender` is

Reply:
I see. Thanks for your advice.

Reply:
I see.  Thanks for your advice.